docs/adding_software/opening_pr.md

```diff
@@ -95,3 +96,13 @@ git push koala example_branch
 
    If all goes well, one or more bots :robot: should almost instantly create a comment in your pull request
    with an overview of how it is configured - you will need this information when providing build instructions.
+
+### Rebuilding software
+We typically do not rebuild software, since (strictly speaking) this breaks reproducibility for anyone using the software. However, there are certain situations in which it is difficult or impossible to avoid.
+
+To do a rebuild, you add the software you want to rebuild to a dedicated easystack file in the `rebuilds` directory. Include the current date (YYYYMMDD) in the name of the EasyStack file. E.g.:
+```shell
+echo '  - example-1.2.3-GCC-12.3.0.eb' >> easystacks/software.eessi.io/2023.06/rebuilds/20240327-eessi-2023.06-eb-4.9.0-2023a.yml
+```
+
+By separating rebuilds in dedicated files, we still maintain a complete software bill of materials: it is transparent what got rebuild, and when.
```

Comment:
```suggestion
By separating rebuilds in dedicated files, we still maintain a complete software bill of materials: it is transparent what got rebuilt, and when.
```

Reply:
And also the reason should be clear, so maybe something like:
```suggestion
By separating rebuilds in dedicated files, we still maintain a complete software bill of materials: it is transparent what got rebuild, for which reason, and when.
```

Reply:
So maybe the paragraph should also mention that, besides the short description in the filename, the easystack file should have some comments at the top explaining why the rebuild is necessary (with links to relevant issues). We could even link to the existing one for OpenMPI as an example?
https://github.com/EESSI/software-layer/blob/2023.06-software.eessi.io/easystacks/software.eessi.io/2023.06/rebuilds/20240301-eb-4.9.0-OpenMPI-4.1.x-fix-smcuda.yml

Reply:
Changed all of the above (but used the CUDA one as example)